### PR TITLE
Rails forms

### PR DESCRIPTION
**Pull request by aliciapaz**
## Rails forms 

Created a simple form to sign up, using different methods: 
- 'Old school' way, with plain HTML
- Using `form_tag` helper method from Rails
- Using `form_with` helper method from Rails

Implemented database-level validations for all fields

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  # def create
+  #   @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+
+  #   if @user.save
+  #     redirect_to new_user_path
+  #   else
+  #     render :new
+  #   end
+  # end
```
Comment (kevcap) [inline app/controllers/users_controller.rb]: Remove commented code.

**Review by kevcap — CHANGES_REQUESTED** [review]
### Hey, nice project so far. We only have to improve a few things before moving on:

Nice job so far, I really liked the effort you put into this project! We just have to do some minor changes so we can approve this project:

I wasn't able to edit a user in the app. 

- [x] it should be possible to edit a user in app ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/edit_user.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

- [x] You have commented code in your users controller, please remove it;


## As always, I wish you all the very best and see you at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (aliciapaz) [reply app/controllers/users_controller.rb]: Thanks for the review.
It is possible to edit the user in <yourlocalhost>/user/<id>/edit

**Comment by aliciapaz** [comment]
Thank you for the review.
It is possible to edit the user in` <localhost>/users/<id>/edit`

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @aliciapaz & @gscarv13**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [ ] in this project you were supposed to create forms in three different ways then use the one that uses form_with and comment out the other ones. please make sure that you add those commented out forms in your new and edit views.


---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+<%
+=end%>
+
+<%
+=begin%>
+ <%= form_tag('/users') do %>
+  <%= label_tag(:username, "Username:") %>
+  <%= text_field_tag(:username) %>
+  <%= label_tag(:email, "Email:") %>
+  <%= email_field(:user, :email) %>
```
Comment (ghost) [inline app/views/users/new.html.erb]: It should be `email_field_tag`. Also, make the change in `edit.html.erb` file.

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉

**Hello team 👋, please check the comment below before you merge. Otherwise, congratulations and good luck with your next project 😃.**

**Pull request merged**